code/modules/species/species_shapeshift.dm

```diff
@@ -554,3 +554,263 @@ var/list/wrapped_species_by_ref = list()
 			H.nutrition -= (2 * nutrition_cost) //Costs Nutrition when damage is being repaired, corresponding to the amount of damage being repaired.
 			H.nutrition = max(0, H.nutrition) //Ensure it's not below 0.
 	..()
+
+
+/mob/living/carbon/human/proc/shapeshifter_copy_appearance()
+	set name = "Mimic Appearance"
+	set category = "Abilities"
+
+	if(stat || world.time < last_special)
+		return
+
+	last_special = world.time + 50 //eh, i'll just leave it as an additional cooldown
+
+	var/list/valid_moblist = list()
+
+	for(var/mob/living/carbon/human/M in oview(7))
+		valid_moblist |= M
+
+	var/mob/living/carbon/human/target = input(src,"Who do you wish to target?","Mimic Target") as null|anything in valid_moblist
+
+	if(!istype(target))
+		return FALSE
+
+	if(target.client.prefs.resleeve_lock)
+		to_chat(src,"<span class='warning'>This person has enabled Prevent Body Impersonation, you cannot copy them!</span>")
+		return FALSE
+
+	if(get_dist(src,target) > 3)
```

Comment:
you build the list from oview(7) but check for dist > 3?
and you should propably change the failure text